pkg/pquerier/frames.go

```diff
@@ -469,16 +469,20 @@ func (d *dataFrame) rawSeriesToColumns() error {
 	nextTime := int64(math.MaxInt64)
 	seriesHasMoreData := make([]bool, len(d.rawColumns))
 
+	emptyMetrics := make(map[int]string)
+
 	for i, rawSeries := range d.rawColumns {
 		if rawSeries == nil {
 			missingColumn := "(unknown column)"
 			for columnName, index := range d.columnByName {
 				if index == i {
```

Comment:
Not new in this PR, but it looks like this could basically just be a map lookup instead of a for loop?

Reply:
Not really, cause you dont have the column name (the map key), you are trying to find it based on the index.